cpp/include/cuml/linear_model/glm.hpp

```diff
@@ -405,85 +272,52 @@ void qnDecisionFunctionSparse(const raft::handle_t& cuml_handle,
  * @param D                     number of features
  * @param C                     number of outputs (C > 1, for multinomial, indicating number of
  * classes. For logistic and normal, C must be 1.)
- * @param fit_intercept         true if model includes a bias.
  * @param params                device pointer to model parameters. Length D if fit_intercept ==
  * false else D+1
- * @param loss_type             id of likelihood model (0: logistic/sigmoid, 1: multinomial/softmax,
- * 2: normal/squared)
  * @param preds                 device pointer to predictions of length N (for binary logistic:
  * [0,1], for multinomial:  [0,...,C-1])
- * @{
  */
+template <typename T>
 void qnPredict(const raft::handle_t& cuml_handle,
-               float* X,
-               bool X_col_major,
-               int N,
-               int D,
-               int C,
-               bool fit_intercept,
-               float* params,
-               int loss_type,
-               float* preds);
-void qnPredict(const raft::handle_t& cuml_handle,
-               double* X,
+               const qn_params& pams,
+               T* X,
                bool X_col_major,
                int N,
                int D,
                int C,
-               bool fit_intercept,
-               double* params,
-               int loss_type,
-               double* preds);
-/** @} */
+               T* params,
+               T* preds);
 
 /**
- * @defgroup qnPredictSparse to fit a GLM using quasi newton methods.
+ * @brief Predict a GLM using quasi newton methods.
+ *
  * @param cuml_handle           reference to raft::handle_t object
- * @param X_values              feature matrix values (CSR format),
- * matrix dimension: NxD.
+ * @param pams                  model parameters
+ * @param X_values              feature matrix values (CSR format), matrix dimension: NxD.
```

Comment:
See other comments about dimensions of the sparse vectors. I do prefer to add the dimensions to the docs.